### PR TITLE
Room component refactor/reorganization to fix some stream issues

### PR DESCRIPTION
**Pull request by ABFranco**
- relocated certain functions into `Room.js` to fix allow streams to be stored in this "parent component" moving forward
  - Includes `RoomForm` and `Vestibule` now being "child" components of `Room`
- Fixed issue with two Join Requests being sent
- gRPC stability is still an issue and not fully investigated in this PR

These changes should help resolve some issues @xmquentin has with Media streams since `Vestibule` is now a child of `Room`.

### CODE REVIEW

File: src/components/Room/Chatroom/Chatroom.js [hunk 1]
```diff
@@ -31,7 +31,14 @@ function Chatroom() {
 
   useEffect(() => {
     // join chat room upon entering the view, this should only occur once
-    joinChatRoomStream();
+    if (store.getState().chatroom.inChatroom) {
```
Comment (xmquentin) [inline src/components/Room/Chatroom/Chatroom.js:34]: ?????

File: src/components/Room/Chatroom/Chatroom.js [hunk 2]
```diff
@@ -31,7 +31,14 @@ function Chatroom() {
...
+      // user is already inChatroom upon a page refresh => leave it
```
Comment (xmquentin) [inline src/components/Room/Chatroom/Chatroom.js]: Cleanup comment please.

File: src/components/Room/Room.js [hunk 3]
```diff
@@ -1,38 +1,57 @@
...
 
+  // initial checks upon loading the page 
   useEffect(() => {
-    // this should only occur once when a non-host
-    if (!store.getState().room.userIsHost) {
-      joinRoomClosureStream();
+
+    // upon refresh, user should always be in the vestibule
+    if (userInRoom) {
+      dispatch(setVestibuleJoin({roomId: store.getState().room.roomId}));
```
Comment (xmquentin) [inline src/components/Room/Room.js:34]: consider adding a bool here for clarity or specifying logic via better commetns

File: src/components/Room/Room.js [hunk 4]
```diff
@@ -1,38 +1,57 @@
...
+    }
+
+    // upon refresh when joining a non-strict room, re-join if not yet accepted
```
Comment (xmquentin) [inline src/components/Room/Room.js]: strict*

File: src/components/Room/Room.js [hunk 5]
```diff
@@ -1,38 +1,57 @@
...
+    if (userInVestibule && store.getState().room.token === null) {
+      roomJoinSubmit(store.getState().vestibule.roomId, store.getState().vestibule.roomPassword, store.getState().vestibule.userName);
     }
   },[]);
 
 
+  // upon entering the room as a non-host, join closure stream
+  useEffect(() => {
+    if (userInRoom && !store.getState().room.userIsHost && store.getState().room.token !== null) {
```
Comment (xmquentin) [inline src/components/Room/Room.js:46]: consider adding more helpful booleans

File: src/components/Room/Room.js [hunk 6]
```diff
@@ -66,24 +83,173 @@ function Room() {
...
+      roomPassword: roomPassword,
+      userName: userName,
+      userId: store.getState().user.userId,
+      isGuest: store.getState().user.userId == null,
+    };
+
+    // reset room data, add userName to state
+    dispatch(clearRoomData());
+    dispatch(clearChatHistory());
+    dispatch(clearVestibuleData());
```
Comment (xmquentin) [inline src/components/Room/Room.js:116]: unnecessary

File: src/components/Room/Room.js [hunk 7]
```diff
@@ -66,24 +83,173 @@ function Room() {
...
+    dispatch(setRoomUserName(userName));
+
+    try {
+      // TODO: relocate joinRoom into vestibuleSlice?
```
Comment (xmquentin) [inline src/components/Room/Room.js]: remove or fix TODO

File: src/components/Room/Room.js [hunk 8]
```diff
@@ -66,24 +83,173 @@ function Room() {
...
+          timestamp: chatHistory[i].getTimestamp(),
+        });
+      }
+      dispatch(setChatHistory(chatHistoryData));
+
+      // update state to allow entering room
+      let payload = {
+        roomId: roomId,
+        token: response.getToken(),
+        isStrict: false, // TODO: does this matter?
```
Comment (xmquentin) [inline src/components/Room/Room.js:193]: pls fix

File: src/components/Room/Room.js [hunk 9]
```diff
@@ -66,24 +83,173 @@ function Room() {
...
+      }
+      dispatch(setJoinedRoom(payload));
+
+    } else if (status === 'wait') {
+      console.log(':Room.receiveJoinRoomResponse: Detected wait room');
+
```
Comment (xmquentin) [inline src/components/Room/Room.js]: unnecessary newlines

File: src/components/RoomForm/RoomCreate/RoomCreate.js [hunk 10]
```diff
@@ -49,7 +50,11 @@ function RoomCreate() {
         dispatch(clearChatHistory());
         dispatch(setRoomUserName(data['userName']));
 
-        history.push(`/vestibule/${store.getState().room.roomId}`);
+        let roomId = store.getState().room.roomId
+        // set data to enter vestibule
```
Comment (xmquentin) [inline src/components/RoomForm/RoomCreate/RoomCreate.js]: unnecessary comment

File: src/components/Room/Room.js (hunk 3 above)
Comment (ABFranco) [reply src/components/Room/Room.js:34]: add a better comment to describe

File: src/components/Room/Room.js (hunk 6 above)
Comment (ABFranco) [reply src/components/Room/Room.js:116]: might need these actually need to investigate

File: src/components/Room/Room.js (hunk 7 above)
Comment (ABFranco) [reply src/components/Room/Room.js]: removing comment b/c outdated

File: src/components/Room/Room.js (hunk 9 above)
Comment (ABFranco) [reply src/components/Room/Room.js]: kk

**Review by xmquentin — APPROVED** [review]

**Pull request merged**